packages/hash/frontend/src/pages/_document.page.tsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import createEmotionServer from "@emotion/server/create-instance";
+import withTwindDocument from "@twind/next/document";
+import Document, { DocumentContext } from "next/document";
+
+import twindConfig from "../../twind.config";
+import { createEmotionCache } from "../theme";
+
+class MyDocument extends Document {
+  static async getInitialProps(context: DocumentContext) {
+    const initialProps = await Document.getInitialProps(context);
+    return { ...initialProps };
+  }
+}
```

Comment:
**Blocking:** we're missing this from [the MUI example](https://github.com/mui/material-ui/blob/master/examples/nextjs/pages/_document.js):
```suggestion
class MyDocument extends Document {
  static async getInitialProps(context: DocumentContext) {
    const initialProps = await Document.getInitialProps(context);
    return { ...initialProps };
  }
  
  render() {
    return (
      <Html lang="en">
        <Head>
          {/* PWA primary color */}
          <meta name="theme-color" content={theme.palette.primary.main} />
          <link rel="shortcut icon" href="/static/favicon.ico" />
          <link
            rel="stylesheet"
            href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
          />
          {/* Inject MUI styles first to match with the prepend: true configuration. */}
          {this.props.emotionStyleTags}
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}
```

Note: I don't think we need the favicon and google font link tags

Reply:
Thanks for catching. Didn't add the `this.props.emotionStyleTags`, because I noticed that already gets added to the page when the page is loaded.

Reply:
Understood, thanks! To be honest I'm not entirely sure why the example adds `this.props.emotionStyleTags`, so if our setup works without I guess this render function isn't strictly necessary right now - but let's keep it as we'll probably want to add to the HTML header here soon anyways.

Reply:
It works in our setup because we attach the tags to the styles key in the object returned by Document.getInitialProps.

We do something like

```
return {
    ...initialProps,
    // Styles fragment is rendered after the app and page rendering finish.
    styles: [
      ...React.Children.toArray(initialProps.styles),
      ...emotionStyleTags,
    ],
```

It seems Nextjs renders styles on the page under the hood. When I tried returning just emotionStyleTags

```
return {
  ...initialProps,
  emotionStyleTags
}
```
The style tags were not added to the page until I explicitly placed `this.props.emotionStyleTags` in the render function